### PR TITLE
feat: filter issues by labels

### LINKED ISSUE #456: Add filtering for issues by labels
Screenshot 2020-01-24 13 07 31

### PR DESCRIPTION
**Pull request by mtfoley**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

This implements issue filtering per issue #456.  I think the UI part still needs work. I decided to add a paragraph explaining the feature, but in order to make the margins look right I used a card, and that put a weird drop shadow above the actual issues list.

## Related Tickets & Documents

Fixes #456 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
Screen Shot 2021-05-01 at 9 50 57 AM

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?
N/A

## [optional] What gif best describes this PR or how it makes you feel?
https://media.giphy.com/media/QZs5EtmpomcMxkGGIc/source.gif

### CODE REVIEW

File: src/components/Issues.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import React, {useState, useEffect} from "react";
-import {FlexCenter} from "../styles/Grid";
+import {Flex, FlexCenter} from "../styles/Grid";
 import {EmptyPlaceholder} from "../styles/EmptyPlaceholder";
 import api from "../lib/apiGraphQL";
 import Card from "./Card";
 import List from "../styles/List";
+import Button from "../styles/Button";
+import Paragraph from "../styles/Paragraph";
+//import Container from "../styles/Container";
```
Comment (0-vortex) [inline src/components/Issues.js:9]: missed this one 😛

File: src/components/Issues.js [hunk 2]
```diff
@@ -71,6 +78,12 @@ function Issues({repoName, owner}) {
       <CardHeader>
         <h1>Issues</h1>
       </CardHeader>
+      <Card style={{paddingBottom: 0, paddingTop: "0.5em"}}>
+        <Flex>
+          <Paragraph>Issues can be filtered by to show good opportunities to assist with this repository.</Paragraph>
```
Comment (0-vortex) [inline src/components/Issues.js:83]: "can be filtered to show" should be used

File: src/styles/Paragraph.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import styled from "styled-components";
+import {size, fontSize} from "./variables";
+
+const Container = styled.p`
```
Comment (0-vortex) [inline src/styles/Paragraph.js:4]: this should be "Paragraph" or "ParagraphContainer" IMO

File: src/components/Issues.js (hunk 2 above)
Comment (mtfoley) [reply src/components/Issues.js:83]: Good catch! I think I was originally going to say something about "filtered by labels". I'll probably change the wording a little more to reflect the fact that the labels are indeed part of the filtering strategy.

File: src/styles/Paragraph.js (hunk 3 above)
Comment (mtfoley) [reply src/styles/Paragraph.js:4]: I don't agree but I only care a little ;-) If the maintainers want it changed, I'll do it.

File: src/components/Issues.js (hunk 1 above)
Comment (mtfoley) [reply src/components/Issues.js:9]: Good catch! my struggles were evident, lol

**Review by bdougie — CHANGES_REQUESTED** [review]
Thanks for taking this on @mtfoley.

Could you move the button next to the title? We can also lose the description text and make that a tooltip. _We don't need to add it in this PR._ 

gfi

**Comment by mtfoley** [comment]
@bdougie should the text read "All Issues" with when the filter is turned on?

Finally, there is one thing I wanted to mention - when I tested this out with a few repos, I noticed that the filter doesn't pick up the cases where an emoji precedes the text.

**Comment by mtfoley** [comment]
new screenshot (also shows the thing about label emojis)
Screen Shot 2021-05-01 at 12 49 15 PM

**Review by bdougie — APPROVED** [review]

**Comment by mtfoley** [comment]
@bdougie I think it would be worth setting up two more docs in onegraph to handle this query with cursors to make next/prev buttons work. See also #971

**Comment by mtfoley** [comment]
This is going to take a little refactoring in Issues.js as the pagination wasn’t made for a changing total issue count, and now that #971 is closed I can dig in. I’ll try to commit some tmrw evening. Might break that pagination footer into its own definition if that helps with state mgmt.

**Comment by mtfoley** [comment]
@bdougie I think those new persisted queries need adjustments (at least the Next page one does, can't test the Previous page one). I feel my latest push (d9bea1b) might be what we need for pagination on the UI, but the persisted query is kicking back an error as is. 

Here's the error from the persisted query:
```js
{
    "errors": [{
        "message": "Variable $cursor is used by Query but not declared",
        "path": ["gitHub"],
        "extensions": {
            "service": "github",
            "upstream": {
                "errors": [{
                    "path": ["query Query", "repositoryOwner", "repository", "issues", "after"],
                    "extensions": {
                        "code": "variableNotDefined",
                        "variableName": "cursor"
                    },
                    "locations": [{
                        "line": 2,
                        "column": 280
                    }],
                    "message": "Variable $cursor is used by Query but not declared"
                }]
            },
            "traceId": "c9ed455c-01a2-4d66-a73e-7e0eacaf7469"
        }
    }],
    "data": {
        "gitHub": null
    }
}
```
If you get new doc_id's for doc_id10 and doc_id11, we can patch them in this PR.

While you've got it open, what do you think of making the labels list a query variable, instead of hard coded? That way we can tweak that part of the feature in the codebase.  If you decide to, I can patch that in the PR as well.

File: src/components/Issues.js [hunk 4]
```diff
@@ -13,16 +14,18 @@ import IssuesLoader from "./IssuesLoader";
...
   const [issuesEnabled, setIssuesEnabled] = useState(null);
   const [offset, setOffset] = useState(0);
-
+  const [totalPages, setTotalPages] = useState(0);
+  const [currentPage, setCurrentPage] = useState(0);
   useEffect(() => {
     setLoading(true);
-    api.persistedRepositoryIssuesFetch(owner, repoName).then(response => {
-      console.log(response)
+    const functionName = issuesFilter ? "persistedIssuesByLabelFetch" : "persistedIssuesFetch";
```
Comment (bdougie) [inline src/components/Issues.js:27]: Instead of `persistedIssuesByLabelFetch` use `persistedRepositoryIssuesByLabelFetch `

**Comment by bdougie** [comment]
@mtfoley new doc ids:

beforeQuery: 407cef3b-eba4-4caa-9a9e-506b25e3f0c9
afterQuery: 90e25769-0a60-4144-8728-ee9e29ccb926

**Comment by mtfoley** [comment]
I pushed with the new doc_id's. OneGraph appears to be janky this morning. I've got a feeling there are some fixes I need to make on pagination but I'm going to see if OneGraph is behaving better later.

**Comment by mtfoley** [comment]
@bdougie I Just tested the deploy preview and motored thru the onegraph errors. I tested it with the repo adafruit/circuitpython because it’s got plenty of good first issues, and with that the paging LGTM. If those onegraph errors stick around, your Friday stream is going to be fun!

**Comment by bdougie** [comment]
I am not seeing the errors. I am comfortable merging this in if you are. The One Graph folks tend to hang out in the chat during the Friday streams, I can also open up a support ticket if the problem arises.

**Comment by mtfoley** [comment]
Looks better now than this morning so I’m good!

**Pull request merged**